### PR TITLE
Raise minimum PHP version to 8.1; add 8.3 to GitHub Actions.

### PR DESCRIPTION
**Pull request by demiankatz**
Now that PHP 8.0 has reached end of life, we should raise our minimum required version. While making changes, we should also add support for PHP 8.3 in continuous integration.

Note: a few tests were failing due to apparent encoding errors in some of our serialized test fixtures. I simply unserialized and reserialized the offending files using an earlier version of PHP, and that seems to have solved the problem.

Note: this includes bare-minimum dependency updates to get things working -- an upgrade to php-cs-fixer (with some associated configuration changes and style fixes) and a migration from the ocramius/proxy-manager package to the more actively maintained friendsofphp/proxy-manager-lts fork.

TODO
- [x] Update changelog when merging
- [x] Update branch protection rules for dev when merging
- [x] Open ticket to change release branch protection rules when releasing 10.0

### CODE REVIEW

**Comment by demiankatz** [comment]
I have confirmed that all Mink tests are passing.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -11,15 +11,15 @@ jobs:
       VUFIND_LOCAL_DIR: $GITHUB_WORKSPACE/local
     strategy:
       matrix:
-        php-version: ['8.0', '8.1', '8.2']
+        php-version: ['8.1', '8.2', '8.3']
         # We run most tests on all platforms, but we only run Javascript-related tests in 8.1.
```
Comment (maccabeelevine) [inline .github/workflows/ci.yaml]: > we only run Javascript-related tests in **8.1**

8.2 now?

Comment (demiankatz) [reply .github/workflows/ci.yaml]: Good catch. I've totally rewritten the comment to be more detailed and future-proof.

**Review by maccabeelevine — APPROVED** [review]
With that comment fix, nothing else stands out, but I'm new to GitHub Actions.  It obviously passes its own tests so that's a good sign. :)

**Comment by demiankatz** [comment]
Thanks, @maccabeelevine! I'll leave this open until next week in case anyone else has anything to add, but I suspect we can probably safely merge this soon.

**Pull request merged**